src/effects/RealtimeEffectList.cpp

```diff
@@ -104,11 +104,37 @@ void RealtimeEffectList::RemoveState(RealtimeEffectState &state)
       mStates.erase(found);
 }
 
+size_t RealtimeEffectList::GetStatesCount() const noexcept
+{
+   return mStates.size();
+}
+
+RealtimeEffectState* RealtimeEffectList::GetStateAt(size_t index) noexcept
```

Comment:
I'm more hesitant about using `noexcept` liberally.  I saw a CppCon talk by John Lakos describing reasons for caution.  I need to review it again.  I mostly use it only when I know something must be called inside a destructor.

I look at this and I see it doesn't have a bounds check on the array access.

Reply:
Given that the index is in bounds (might say it with a `@pre` comment), should this return a reference to a guaranteed non-null object?  Perhaps.  But maybe warn too about invalidation in case states are added or removed.

Reply:
Added a commetary and replaced pointer with reference type